crates/ide/src/runnables.rs

```diff
@@ -192,6 +192,44 @@ pub(crate) fn runnables(db: &RootDatabase, file_id: FileId) -> Vec<Runnable> {
     res
 }
 
+// Feature: Selected Test
+//
+// Finds the matching test that contains the provided position.
+//
+// This feature is used when attempting to run or debug a selected test to choose
+// the test to run. Super useful for quickly running just a single test. Do bind
+// this to a shortcut!
+//
+// |===
+// | Editor  | Action Name
+//
+// | VS Code | **rust-analyzer: Run Selected**
+// | VS Code | **rust-analyzer: Debug Selected**
+// |===
+pub(crate) fn selected_test(db: &RootDatabase, position: FilePosition) -> Option<Runnable> {
+    let sema = Semantics::new(db);
+    let syntax = sema.parse(position.file_id).syntax().clone();
+    return find_selected_test(&sema, &syntax, position);
```

Comment:
`.clone()` seems unnecessary cause `find_selected_test` takes a `&syntax`

Reply:
This is right. I copied this process from `related_tests` so I'll give that an update too.

Reply:
squash commits would be nice

Reply:
you can run `cargo test` to verify that locally